cpp/include/cuspatial/detail/geometry/polygon_ref.cuh

```diff
@@ -59,13 +59,13 @@ CUSPATIAL_HOST_DEVICE auto polygon_ref<RingIterator, VecIterator>::ring_end() co
 template <typename RingIterator, typename VecIterator>
 CUSPATIAL_HOST_DEVICE auto polygon_ref<RingIterator, VecIterator>::point_begin() const
 {
-  return _point_begin;
+  return thrust::next(_point_begin, *_ring_begin);
```

Comment:
Can you explain these changes? Why can't point_begin be used as stored?

Reply:
tl;dr. Because the offsets stored in a `*_ref` are *global* offsets to the lower level range.

Consider this example:
Multilinestring_range:
Geometry offsets: [0, 2, 4]
Part Offsets: [0, 2, 4, 6, 8]
Points: [P0, P1... P8]

A `multilinestring_ref` constructed for the second multilinestring is stored as below:
Part Offsets: [4, 6, 8]
Points: [P0, ... P8]

Notice that in part offset, the offsets are global offsets (4 means that the first point to the multilinestring locates at _point_begin + 4). You may ask why we don't store as below:

Part Offsets: [0, 2, 4]
Points: [P4 ... P8]

Because this requires a scan to the part offset and could be expensive.